File: javascript/phone-number/phone-number.js
export class PhoneNumber {
  constructor(str) {
    const re = new RegExp(/^(?:\+?1? *\(?)([2-9]\d{2})(?:\)? *\.?)([2-9]\d{2})(?:-?\.? *)(\d{4})(?:\W*)$/);
    const matches = str.match(re);
    try {
      const trial = `${matches[1]}${matches[2]}${matches[3]}`;
      this.cleaned = trial.length === 10 ? trial : null;
    }
    catch (e) {
      this.cleaned = null;
    }
  }

  number() {
    return this.cleaned;
  }
}

File: python/phone-number/phone_number.py
import re

class PhoneNumber:
    def __init__(self, number):
        regex = r"\d+"
        matches = re.findall(regex, number)
        num = ''
        for match in matches:
            num += str(match)
        if num[0] == '1':
            num = num[1:]
        if len(num) != 10 or num[0] == '1' or num[0] == '0' or num[3] == '1' or num[3] == '0':
            raise ValueError('Invalid number')
        self.__num = num
        self.__area_code = num[:3]
        self.__exchange = num[3:6]
        self.__last_four = num[6:]

    @property
    def number(self):
        return self.__num

    @property
    def area_code(self):
        return self.__num[:3]

    def pretty(self):
        return "({}) {}-{}".format(self.__area_code, self.__exchange, self.__last_four)
